lib/utils/metrics/snmp-pdu-sensor-status.js

```
@@ -0,0 +1,112 @@
+        .then(self.collectSensorData.bind(self))
+        .then(self.calculateSensorData.bind(self));
+    };
+
+    SnmpPduSensorMetric.prototype.collectSensorData = function() {
+        var self = this;
+        if (this.nodeType === 'sinetica') {
+            return Promise.all([
+                    self._collectSineticaSensorData(['HAWK-I2-MIB::ipTHATable']),
+                    self._collectSineticaSensorData(['HAWK-I2-MIB::ipContTable'])
+                ]
+                )
+                .then(function (results) {
+                    for (var i = (results.length)-1; i >= 0; i-- ) {
+                        if (results[i] === undefined) {
+                            results.splice(i, 1);
+                        }
+                    }
+                    return (results);
+                });
+        }
+    };
+
+    SnmpPduSensorMetric.prototype.calculateSensorData = function(values) {
+        var self = this;
+        var out={};
+        var data;
+        if (this.nodeType === 'sinetica') {
+            _.forEach(values, function (value) {
+                var source = value.source;
+                source = source.substring(source.lastIndexOf(":") + 1,
+                    source.indexOf("Table"));
+                data= (self._calculateSineticaSensorData(value.values, source));
+                var keys = Object.keys(data);
+                for(var i = 0 ;i< keys.length; i++) {
+                    out[keys[i]] = data[keys[i]];
+                }
+            });
+        }
+        return out
+    };;
+
+    SnmpPduSensorMetric.prototype._collectSineticaSensorData = function(pduOids) {
+        var self = this;
+        var snmpQueryType = 'bulkwalk';
+        return self.snmptool.collectHostSnmp(pduOids, {
+            snmpQueryType: snmpQueryType,
+               isSequential: true
+        })
+        .then(function (pduOidsResult) {
+            return pduOidsResult[0];
+
+        }).catch(function(err) {
```

This `catch` is just want to ignore any error while collecting sensor data?